packages/react-charting/src/components/Legends/Legends.base.tsx

```diff
@@ -392,26 +420,26 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
     const { theme } = this.props;
     const { palette } = theme!;
     let legendColor = color;
-    const inSelectedState = this.state.selectedLegend !== '' || this.state.selectedLegends.length > 0;
-    if (inSelectedState) {
-      /** if one or more legends are selected */
-      if (this.state.selectedLegend === title || this.state.selectedLegends.indexOf(title) > -1) {
-        /** if the given legend (title) is one of the selected legends */
+    // if one or more legends are selected
+    if (this.state.isLegendSelected) {
```

Comment:
isLegendSelected seems to be a derived variable and used only once. Can we use an expression from the other variables to compute this value

Reply:
We only need to know if the selectedLegends object is not empty. But to check that, we must find the number of keys, which takes O(n) time.
https://stackoverflow.com/questions/7716812/object-keys-complexity